runtime/util/ConfigurationSerializer.cpp

```diff
@@ -420,3 +420,45 @@ void serializeConfiguration(
   *data_out = buf;
   *size_out = size;
 }
+
+void writeUInt64ToFile(const char *filename, uint64_t i) {
+  FILE *file = fopen(filename, "a");
+  fwrite(&i, 8, 1, file);
+  fclose(file);
+}
+
+void serializeTermToFile(
+    const char *filename, block *subject, const char *sort) {
+  char *data;
+  size_t size;
+  serializeConfiguration(subject, sort, &data, &size, false);
+
+  FILE *file = fopen(filename, "a");
+  fwrite(data, 1, size, file);
+  fclose(file);
+}
+
+void serializeRawTermToFile(
```

Comment:
This function ends up duplicating some of the logic from `debug_print_term` in `ConfigurationPrinter.cpp` - in particular, it does the same work of constructing a term injected into `KItem`.

Could we possibly extract the common part of these two functions?

n.b. I believe your implementation demonstrates a subtle bug in mine when the term's sort is `Bool` or `MInt{...}`!

Reply:
What happens to my code if you unconditionally add the indirection? If it still _works_ but does some extra work, I'd say drop the flag

Reply:
It gives the boolean value of true where false is expected in one of the tests. Presumably because the non-zero pointer to the boolean value is read as the boolean value itself.